apis/core/lib/domain/csv-source/update.ts

```diff
@@ -3,23 +3,25 @@ import { NamedNode } from 'rdf-js'
 import { CsvSource } from '@cube-creator/model'
 import RdfResource from '@tpluscode/rdfine'
 import { schema } from '@tpluscode/rdf-ns-builders'
+import type { MediaObject } from '@rdfine/schema'
 import { ResourceStore } from '../../ResourceStore'
 import { error } from '../../log'
-import * as s3 from '../../storage/s3'
 import { loadFileHeadString } from '../csv/file-head'
 import { parse } from '../csv'
 import { sampleValues } from '../csv/sample-values'
+import type { MediaStorage } from '../../storage'
+import { getMediaStorage } from '../../storage'
 
 interface UpdateCsvSourceCommand {
   resource: GraphPointer<NamedNode>
   store: ResourceStore
-  fileStorage?: s3.FileStorage
+  getStorage?: (m: MediaObject) => MediaStorage
```

Comment:
I see this used a bunch of times. You may find it useful to export a callable interface

```typescript
export interface GetMediaStorage {
  (m: MediaObject) => MediaStorage
}
```

Reply:
I kinda feel like specifying the whole function is more readable... but maybe not